tradeexecutor/ethereum/uniswap_v3_execution.py

```diff
@@ -0,0 +1,199 @@
+"""Execution model where trade happens directly on Uniswap v2 style exchange."""
+
+import datetime
+from decimal import Decimal
+from typing import List
+import logging
+
+from web3 import Web3
+
+from eth_defi.hotwallet import HotWallet
+from tradeexecutor.ethereum.execution import wait_trades_to_complete
+from tradeexecutor.ethereum.execution_uniswap_v3 import broadcast_and_resolve, resolve_trades
+from tradeexecutor.ethereum.uniswap_v3_routing import UniswapV3SimpleRoutingModel, UniswapV3RoutingState
+from tradeexecutor.state.freeze import freeze_position_on_failed_trade
+from tradeexecutor.state.state import State
+from tradeexecutor.state.trade import TradeExecution, TradeStatus
+from tradeexecutor.strategy.execution_model import ExecutionModel
+
+
+logger = logging.getLogger(__name__)
+
+
+class UniswapV3ExecutionModel(ExecutionModel):
```

Comment:
We can get away with the same execution model  for Uniswap v2 and v3, so it is unlikely we need to have a new class.

This code needs to be rethought,